mlflow/utils/credentials.py

```diff
@@ -41,3 +46,130 @@ def read_mlflow_creds() -> MlflowCreds:
         username=username_env or username_file,
         password=password_env or password_file,
     )
+
+
+def login(backend="databricks"):
+    """Configure MLflow server authentication and connect MLflow to tracking server.
+
+    This method provides a simple way to connect MLflow to its tracking server. Currently only
+    Databricks tracking server is supported. Users will be prompted to enter the credentials if no
+    existing Databricks profile is found, and the credentials will be saved to `~/.databrickscfg`.
+
+    Args:
+        backend: string, the backend of the tracking server. Currently only "databricks" is
+            supported.
+
+    .. code-block:: python
+        :caption: Example
+
+        import mlflow
+
+        mlflow.login()
+        with mlflow.start_run():
+            mlflow.log_param("p", 0)
+    """
+    if backend == "databricks":
+        _databricks_login()
+    else:
+        raise ValueError(
+            f"Currently only 'databricks' backend is supported, received `backend={backend}`."
+        )
+
+
+def _check_databricks_auth():
+    # Check if databricks credentials are set.
+    from databricks.sdk import WorkspaceClient
+
+    try:
+        w = WorkspaceClient()
+        # If credentials are invalid, `clusters.list()` will throw an error.
+        w.clusters.list()
+        _logger.info(
+            "Succesfully signed in Databricks! Please run `mlflow.set_tracking_uri('databricks')` "
+            "to connect MLflow to Databricks tracking server."
+        )
+        return True
+    except Exception:
+        return False
+
+
+def _overwrite_or_create_databricks_profile(
+    file_name,
+    profile,
+    profile_name="DEFAULT",
+):
+    """Overwrite or create a profile in the databricks config file.
+
+    Args:
+        file_name: string, the file name of the databricks config file, usually `~/.databrickscfg`.
+        profile: dict, contains the authentiacation profile information.
+        profile_name: string, the name of the profile to be overwritten or created.
+    """
+    profile_name = f"[{profile_name}]"
+    lines = []
+    # Read `file_name` if the file exists, otherwise `lines=[]`.
+    if os.path.exists(file_name):
+        with open(file_name) as file:
+            lines = file.readlines()
+    start_index = -1
+    end_index = -1
+    # Find the start and end indices of the profile to overwrite.
+    for i in range(len(lines)):
+        if lines[i].strip() == profile_name:
+            start_index = i
+            break
+
+    if start_index != -1:
+        for i in range(start_index + 1, len(lines)):
+            # Reach an empty line or a new profile.
+            if lines[i].strip() == "" or lines[i].startswith("["):
+                end_index = i
+                break
```

Comment:
```suggestion
    enum_lines = enumerate(lines)
    for idx, line in enum_lines:
        if line.strip() == profile_name:
            start_index = idx
            break

    if start_index != -1:
        for idx, line in enum_lines:
            # Reach an empty line or a new profile.
            if line.strip() == "" or line.startswith("["):
                end_index = idx
                break
```

Reply:
With this proposal, line 124 will reuse the iterator defined on line 117, which might be a bit confusing. The old code is a bit verbose, but is easier to read IMO.

Reply:
Got it, can we consider using `enumerate`?

```suggestion
    for idx in enumerate(lines):
        if line.strip() == profile_name:
            start_index = idx
            break

    if start_index != -1:
        for idx, line in enumerate(lines[start + 1:], start + 1):
            # Reach an empty line or a new profile.
            if line.strip() == "" or line.startswith("["):
                end_index = idx
                break
```